### ISSUE TITLE
"Convert to ES6 module" not working with IIFE pattern

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#24297]
_From @osya on May 21, 2018 8:59_

- VSCode Version: 1.23.1
- OS Version: Windows 10 Pro x64

Steps to Reproduce:

1. Create a file test_file.js with the following content:
```
var path = require('path');

(function () {
    var path = require('path');
}());
```
![scr](https://user-images.githubusercontent.com/998657/40297920-14f5fae2-5cea-11e8-86d1-cc4261470383.png)
Click lightbulb and select "Convert to ES6 module" then the 1st `require` transforms to the `import path from "path";`. It is correct. Then comment 1st `require` and click lightbulb and select "Convert to ES6 module" for the 2nd `require`. 2nd `require` doesn't converted to the `import path from "path";`. ES6 imports must be at the top level of your JavaScript files. So, I suppose, for `require` in IIFE there should be NO option "Convert to ES6 module"

_Copied from original issue: Microsoft/vscode#50205_

**Comment by mjbvz** [comment]
_From @vscodebot[bot] on May 21, 2018 8:59_

(Experimental duplicate detection)
Thanks for submitting this issue. Please also check if it is already covered by an existing one, like:
- ["convert to a ES6 module" code action doesn't preserve type of quotes (#46679)](https://www.github.com/Microsoft/vscode/issues/46679)

**Comment by mjbvz** [comment]
Yes, I think not converting the inner `require` is correct since imports can only appear at the top level in a program. 

In VS Code insiders, I do not see the `convert to es6` suggestion on the inner require:

![screen shot 2018-05-21 at 11 31 39 am](https://user-images.githubusercontent.com/12821956/40323560-8eba2b96-5cea-11e8-99aa-a5681b0a895f.png)

**Comment by osya** [comment]
@mjbvz , may be you just need to wait a little bit. There is a screenshot with existing "Convert to ES6 module" suggestion for `require` in the IIFE 
![scr](https://user-images.githubusercontent.com/998657/40325030-102faf7c-5d43-11e8-9a21-41c17eec25af.png)

**Comment by DanielRosenwasser** [comment]
(As an aside, could you tell us a bit more about why this IIFE is being used?)

**Comment by mhegazy** [comment]
@osya this last report should be fixed by https://github.com/Microsoft/TypeScript/pull/24101.

**Comment by osya** [comment]
@DanielRosenwasser I use IIFE for the following two reasons:
1. I `'use strict';`. Then without IIFE `jshint` generates warning: "Use the function form of "use strict". (W097)". So, the 1st reason for using IIFE is to avoid this warning. 
2. [Resharper's warning "Property or global variable ... is possibly never assigned"](http://stackoverflow.com/questions/7604419/resharper-javascript-use-of-implicitly-declared-global-variable-x)

**Comment by andy-ms** [comment]
@osya This was fixed recently in #24101, you may need to install `typescript@next` to stop seeing the suggestion in inner scopes.
